### ISSUE TITLE
[NPU][ResNet50]  INT64 数据类型跑在 AI_CPU 上而不是 AI_Core 上

### ISSUE THREAD

**Issue by qili93** [PaddlePaddle/PaddleCustomDevice#184]
**表现：** Mul, Cast 等输入数据类型为 INT64 的算子跑在 AI_CPU 而不是 AI_CORE 上
**原因：** 昇腾等系列国产硬件对 INT64 数据类型支持有问题，但是飞桨框架默认数据集的 Label 读取为 int64 类似
**解决办法：** 修改套件代码，强制返回 Label 数据类型为 int32 类型
**分析方法：** 模型中添加 acl profiler 接口获得性能分析结果

**临时解决方案：**

```bash
# 模型代码中调用 mul, cast 等算子的代码为
cost = nn.CrossEntropyLoss()
loss = cost(outputs, labels) # 这里调用

# 框架代码中调用 mul, cast 等算子的代码为
def cross_entropy(... ...
valid_label = paddle.multiply(paddle.cast(label != ignore_index, dtype=label.dtype), label)

# 解决办法，修改 diff 如下，或者修改套件代码里面的 dataset 代码
diff --git a/python/paddle/vision/datasets/folder.py b/python/paddle/vision/datasets/folder.py
index 6ac0c4ca91..569396b482 100644
--- a/python/paddle/vision/datasets/folder.py
+++ b/python/paddle/vision/datasets/folder.py
@@ -271,8 +271,10 @@ class DatasetFolder(Dataset):
         sample = self.loader(path)
         if self.transform is not None:
             sample = self.transform(sample)
+        import numpy as np
+        return sample, np.array([target]).astype('int32')

-        return sample, target
+        #return sample, target

     def __len__(self):
         return len(self.samples)
```
